### ISSUE TITLE
Rendering Pass to Framebuffer Object

### ISSUE THREAD

**Issue by AndrewJo** [Rajawali/Rajawali#384]
Continuing the discussion from #383 with @alenko and @lacasrac, I think enough people want to be able to apply post processing filters to some of the elements as opposed to the entire scene. This basically means there should be a more flexible way to render to a framebuffer object.

Probably would be best if this gets labeled as discussion and enhancement?

I'll attach a step-by-step render-to-texture pseudocode below to start off.
## Render to Texture Pseudocode

Referenced and "Javafied" from OpenGL ES 2.0 Programming Guide

``` java
// Generate framebuffer, renderbuffer, and texture
GLES20.glGenFramebuffers(1, frameBuffer);
GLES20.glGenRenderbuffers(1, depthRenderBuffer);
GLES20.glGenTextures(1, texture);

// Bind and load texture mipmap level 0 and use RGB 565 mode
GLES20.glBindTexture(GL_TEXTURE_2D, texture);
GLES20.glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, texWidth, texHeight, 0, GL_RGB, GL_UNSIGNED_SHORT_5_6_5, NULL);

// Texture wrap and filter parameters
GLES20.glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
GLES20.glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
GLES20.glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
GLES20.glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);

// Bind renderbuffer
// Width and height of the renderbuffer = width and height of texture
GLES20.glBindRenderbuffer(GL_RENDERBUFFER, depthRenderbuffer);
GLES20.glRenderbufferStorage(GL_RENDERBUFFER, GL_DEPTH_COMPONENT16, texWidth, texHeight);

// Bind framebuffer
GLES20.glBindFramebuffer(GL_FRAMEBUFFER, framebuffer);

GLES20.glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, texture, 0);
GLES20.glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_RENDERBUFFER, depthRenderBuffer);

// Check framebuffer status
status = glCheckFramebufferStatus(GL_FRAMEBUFFER);

// Render to texture using framebuffer object (FBO) when correctly setup
if (status == GL_FRAMEBUFFER_COMPLETE) {
    // First clear color and depth buffer
    GLES20.glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
    GLES20.glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);

    // Setup the shaders and load uniforms here
    ...

    // Draw
    objectFoo.render();

    // Render to window system provided by framebuffer
    GLES20.glBindFramebuffer(GL_FRAMEBUFFER, 0);

    // Draw to the quad that is the size of the viewport here
    // Refer to PostProcessingRenderer on how it is accomplished in Rajawali
    ...
}

// Cleanup code
GLES20.glDeleteRenderbuffers(1, depthRenderBuffer);
GLES20.glDeleteFramebuffers(1, frameBuffer);
GLES20.glDeleteTextures(1, texture);
```

@alenko mentioned steps to achieve selective post processing here: [#383](https://github.com/MasDennis/Rajawali/pull/383#issuecomment-13430831) but I don't know how feasible this is. Looking at the underlying source code, we will most likely break backwards compatibility trying to implement this feature.
## A Usecase

Basically the steps for glow/light bloom would be:
1. Render glowmap in the first pass to a framebuffer
2. Blur the rendered glowmap by applying the BlurFilter
3. Blend the glowmap with the fully rendered scene
